### PR TITLE
Improve entities and DTOs management

### PR DESCRIPTION
**Pull request by ludelafo**

### CODE REVIEW

File: src/auth/auth.service.ts [hunk 1]
```diff
@@ -67,16 +68,16 @@ export class AuthService {
...
-	async validateSpot(spotId: string) {
+	async validateSpot(spotId: string): Promise<User> {
 		const spot = await this.spotsService.getSpot(spotId);
 
 		if (spot.configured) {
 			throw new UnauthorizedException();
 		}
 
-		const user = (await this.usersService.getUser(spot.userId)) as User;
+		const spotWithUser = await this.spotsService.getSpotWithUser(spot.id);
```
Comment (Nortalle) [inline src/auth/auth.service.ts:78]: Nice touch !

File: src/users/enums/user-role.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export enum UserRole {
```
Comment (Nortalle) [inline src/users/enums/user-role.ts:1]: Should it be used in seed.ts ?

**Review by Nortalle — CHANGES_REQUESTED** [review]

File: src/users/enums/user-role.ts (hunk 2 above)
Comment (ludelafo) [reply src/users/enums/user-role.ts:1]: As discussed, the `src` directory will not be copied in the Docker image so we can't reference it from the `seed.ts` file. I think we are obliged to use as it is now.

File: public/javascripts/script.js [hunk 3]
```diff
@@ -1,40 +1,90 @@
...
+			li.className = 'error';
+			li.innerHTML = message[0].toUpperCase() + message.slice(1);
+
+			ul.appendChild(li);
+		}
+	} else if (response.status === 401) {
+		const li = document.createElement('li');
+
+		li.className = 'error';
+		li.innerHTML = 'You are not authorized to perform this action.';
```
[the remark this answers is not in the document]
Comment (ludelafo) [reply public/javascripts/script.js:77]: Good point. Let's add a branch for 403 errors and improve the current message.

**Review by Nortalle — APPROVED** [review]

**Pull request merged**